### DISCUSSION TITLE
What's the recommended way to manage flymake-diagnostic-functions for a specific mode?

### DISCUSSION THREAD

**Discussion by Cerebus** [joaotavora/eglot#1122] [Q&A] [answered]
Specifically, pyright doesn't do linting, so I want to keep `python-flymake` in the diagnostics functions list.  Currently I'm doing this (adapted from a reddit thread):
```
(defun my/eglot-python-fixup ()
  (add-hook 'eglot-managed-mode-hook (lambda () (add-hook 'flymake-diagnostic-functions #'python-flymake t t)))
  )

(add-hook 'python-mode-hook #'my/eglot-python-fixup)
```

This seems to work but I haven't tested it across multiple editing modes and sessions, and I'm not sure it's the best approach.  Anyone have a better idea?

**Comment by joaotavora** [comment, marked as answer]
This is exactly the right way to do it at the moment. At least until Eglot grows support for multiple servers, or until pyright grows support for additional checkers. The latter is what the clangd C++ server does: it has multiple checkers on a single server.

**Comment by joaotavora** [comment]
Another way is to add directly to eglot-managed-mode-hook and check the value of major-mode there. Could be a bit less convoluted.
